clang/include/clang/Sema/Sema.h

```diff
@@ -1909,7 +1909,19 @@ class Sema final : public SemaBase {
   /// '\#pragma clang attribute push' directives to the given declaration.
   void AddPragmaAttributes(Scope *S, Decl *D);
 
-  void PrintPragmaAttributeInstantiationPoint();
+  using DiagFuncRef =
+      llvm::function_ref<void(SourceLocation, PartialDiagnostic)>;
+  auto getDefaultDiagFunc() {
+    return [this](SourceLocation Loc, PartialDiagnostic PD) {
+      DiagnosticBuilder Builder(Diags.Report(Loc, PD.getDiagID()));
+      PD.Emit(Builder);
+    };
+  }
```

Comment:
I think we could / should put that into SemaTemplateInstantiate - and have the function taking a  DiagFuncRef have that default parameter defaulted to `{}`

Reply:
I thought about / tried that first, but assuming multiple users, I think there is less duplication this way, and the duplication is more obvious.
There are some other functions that can start taking this parameter in future patches.